docs/data/joy/components/select/SelectFieldDemo.js

```diff
@@ -9,20 +9,26 @@ export default function SelectFieldDemo() {
   const [value, setValue] = React.useState('dog');
   return (
     <Box sx={{ width: 240 }}>
-      <FormLabel>Favorite pet</FormLabel>
+      <FormLabel
+        id="select-field-demo-label"
+        htmlFor="select-field-demo-button"
+        sx={{ mb: 0.25 }}
+      >
+        Favorite pet
+      </FormLabel>
       <Select
         defaultValue="dog"
         componentsProps={{
           button: {
+            id: 'select-field-demo-button',
             // screen readers will announce "Favorite pet, dog selected" when the select is focused.
-            'aria-label': `Favorite pet, ${value} selected.`,
+            'aria-labelledby': 'select-field-demo-label select-field-demo-button',
```

Comment:
I think that `aria-labelledby` > `aria-label` in this context because it makes it possible for developers that copy this demo to leave the component uncontrolled. See how the dependency on `value` is gone.